cmd/lock-rpc-client.go

```diff
@@ -0,0 +1,71 @@
+/*
+ * Minio Cloud Storage, (C) 2016 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import "github.com/minio/dsync"
+
+// LockRPCClient is authenticable lock RPC client compatible to dsync.NetLocker
+type LockRPCClient struct {
+	*AuthRPCClient
+}
+
+// newLockRPCClient returns new lock RPC client object.
+func newLockRPCClient(config authConfig) *LockRPCClient {
+	return &LockRPCClient{newAuthRPCClient(config)}
+}
+
+// RLock calls read lock RPC.
```

Comment:
It should reconnect dsync doesn't reconnect on its own, dsync just calls the same lock on the resource.  That is what the retry does in `defaultRetryConfig`